torch/csrc/jit/register_prim_ops.cpp

```diff
@@ -3400,16 +3400,36 @@ at::Tensor interpolate(
     }
   }
 
+  float scale_factors_1 = -1;
+  float scale_factors_2 = -1;
+  float scale_factors_3 = -1;
+  if (scale_factors.isDouble()) {
+    scale_factors_1 = (float) scale_factors.toDouble();
```

Comment:
Not clear on why we need to convert explicitly to double using `.toDouble` if it is already a double (as checked in the line above). 
Also, for casting to float, can we use the modern `static_cast<float>`?

Reply:
scale_factors is not a double, it is an instance of IValue.
toDouble() is a function of the class that returns the value of the double.